From a1151abc5d3aab9c9a4118673b139bfbea68b87a Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 13 Jun 2022 12:00:32 +0100
Subject: [PATCH] overlays: Delete+deprecate the rpivid-v4l2 overlay

Now that the rpivid-vid-decoder driver is the default, this overlay
is no longer needed.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/Makefile           |  1 -
 arch/arm/boot/dts/overlays/README             |  7 ++-
 arch/arm/boot/dts/overlays/overlay_map.dts    |  2 +-
 .../boot/dts/overlays/rpivid-v4l2-overlay.dts | 50 -------------------
 4 files changed, 4 insertions(+), 56 deletions(-)
 delete mode 100644 arch/arm/boot/dts/overlays/rpivid-v4l2-overlay.dts

diff --git a/arch/arm/boot/dts/overlays/Makefile b/arch/arm/boot/dts/overlays/Makefile
index f6bc9a84af5e..14b855f648ed 100644
--- a/arch/arm/boot/dts/overlays/Makefile
+++ b/arch/arm/boot/dts/overlays/Makefile
@@ -185,7 +185,6 @@ dtbo-$(CONFIG_ARCH_BCM2835) += \
 	rpi-proto.dtbo \
 	rpi-sense.dtbo \
 	rpi-tv.dtbo \
-	rpivid-v4l2.dtbo \
 	rra-digidac1-wm8741-audio.dtbo \
 	sainsmart18.dtbo \
 	sc16is750-i2c.dtbo \
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index 8ccace1511d8..365005982d80 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -3141,10 +3141,9 @@ Params: <None>
 
 
 Name:   rpivid-v4l2
-Info:   Load the V4L2 stateless video decoder driver for the HEVC block,
-        disabling the memory mapped devices in the process.
-Load:   dtoverlay=rpivid-v4l2
-Params: <None>
+Info:   This overlay has been deprecated and deleted as the V4L2 stateless
+        video decoder driver is enabled by default.
+Load:   <Deprecated>
 
 
 Name:   rra-digidac1-wm8741-audio
diff --git a/arch/arm/boot/dts/overlays/overlay_map.dts b/arch/arm/boot/dts/overlays/overlay_map.dts
index 0e01f46d8db0..ade9f488bdfd 100644
--- a/arch/arm/boot/dts/overlays/overlay_map.dts
+++ b/arch/arm/boot/dts/overlays/overlay_map.dts
@@ -62,7 +62,7 @@ pi3-miniuart-bt {
 	};
 
 	rpivid-v4l2 {
-		bcm2711;
+		deprecated = "no longer necessary";
 	};
 
 	sdio-1bit {
diff --git a/arch/arm/boot/dts/overlays/rpivid-v4l2-overlay.dts b/arch/arm/boot/dts/overlays/rpivid-v4l2-overlay.dts
deleted file mode 100644
index bdd1c0e5a915..000000000000
--- a/arch/arm/boot/dts/overlays/rpivid-v4l2-overlay.dts
+++ /dev/null
@@ -1,50 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-only
-// Definitions for Raspberry Pi video decode engine
-/dts-v1/;
-/plugin/;
-
-#include <dt-bindings/interrupt-controller/arm-gic.h>
-
-/{
-	compatible = "brcm,bcm2711";
-
-	fragment@0 {
-		target = <&scb>;
-		__overlay__ {
-			/* needed to avoid dtc warning */
-			#address-cells = <2>;
-			#size-cells = <2>;
-
-			codec@7eb10000 {
-				compatible = "raspberrypi,rpivid-vid-decoder";
-				reg = <0x0 0x7eb10000  0x0 0x1000>,  /* INTC */
-				      <0x0 0x7eb00000  0x0 0x10000>; /* HEVC */
-				reg-names = "intc",
-					    "hevc";
-
-				interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
-
-				clocks = <&firmware_clocks 11>;
-				clock-names = "hevc";
-			};
-		};
-	};
-
-	fragment@1 {
-		target = <&scb>;
-		__overlay__ {
-			hevc-decoder@7eb00000 {
-				status = "disabled";
-			};
-			rpivid-local-intc@7eb10000 {
-				status = "disabled";
-			};
-			h264-decoder@7eb20000 {
-				status = "disabled";
-			};
-			vp9-decoder@7eb30000 {
-				status = "disabled";
-			};
-		};
-	};
-};
-- 
2.32.0

